smart_home/include/devices/ac_unit.h

```diff
@@ -0,0 +1,53 @@
+#ifndef SMART_HOME_DEVICES_AC_UNIT_H_
+#define SMART_HOME_DEVICES_AC_UNIT_H_
+
+#include "../rooms/room.h"
+#include "device.h"
+
+namespace devices {
+
+// Rule of Five
+
+class AcUnit : public Device {
+  public:
+    AcUnit(rooms::Room* room): Device(room), is_on_(false), fan_speed_(0) {}
+
+    rooms::Room* room() override {
+        return room_;
+    }
+
+    inline bool is_on() {
+        return is_on_;
+    }
+
+    inline void set_is_on(bool value) {
+        is_on_ = value;
+    }
+
+    inline int fan_speed() {
+        return fan_speed_;
+    }
+
+    inline void set_fan_speed(int value) {
+        fan_speed_ = value;
+    }
+
+    inline float desired_temperature() {
+        return desired_temperature_;
+    }
+
+    inline void set_desired_temperature(float value) {
+        desired_temperature_ = value;
+    }
```

Comment:
observe the Google Style naming convention: https://google.github.io/styleguide/cppguide.html#Function_Names
and also read more about const functions